RunFullAnalysis.sh

```diff
@@ -218,17 +257,53 @@ fi
+      echo $(tput setaf 4) Compute efficiency from ${OutDirEfficiency}/Distr_${Particle}_MC${CutSets[$iCutSet]}.root $(tput sgr0)
+      python3 ComputeEfficiencyDplusDs.py ${cfgFileFit} ${Cent} ${OutDirEfficiency}/Distr_${Particle}_MC${CutSets[$iCutSet]}.root ${OutDirEfficiency}/Efficiency_${Particle}${CutSets[$iCutSet]}.root --batch
+    fi
   done
 fi
 
 #compute efficiency times acceptance
 if $DoAccEff; then
   for (( iCutSet=0; iCutSet<${arraylength}; iCutSet++ ));
   do
-    echo $(tput setaf 4) Compute efficiency times acceptance $(tput sgr0)
-    python3 CombineAccTimesEff.py ${OutDirEfficiency}/Efficiency_${Particle}${CutSets[$iCutSet]}.root ${accFileName} ${OutDirEfficiency}/Eff_times_Acc_${Particle}${CutSets[$iCutSet]}.root --batch
+    if [ ${Particle} == "LctopKpi" ]; then
+      echo $(tput setaf 4) Compute efficiency times acceptance $(tput sgr0)
+      python3 CombineAccTimesEff.py ${OutDirEfficiency}/Efficiency_${Particle}${CutSets[$iCutSet]}.root ${accFileName} ${OutDirEfficiency}/Eff_times_Acc_${Particle}${CutSets[$iCutSet]}.root --batch
+      python3 CombineAccTimesEff.py ${OutDirEfficiency}/Efficiency_${Particle}_NonRes${CutSets[$iCutSet]}.root ${accFileName} ${OutDirEfficiency}/Eff_times_Acc_${Particle}_NonRes${CutSets[$iCutSet]}.root --batch
+      python3 CombineAccTimesEff.py ${OutDirEfficiency}/Efficiency_${Particle}_KStar${CutSets[$iCutSet]}.root ${accFileName} ${OutDirEfficiency}/Eff_times_Acc_${Particle}_KStar${CutSets[$iCutSet]}.root --batch
+      python3 CombineAccTimesEff.py ${OutDirEfficiency}/Efficiency_${Particle}_Delta${CutSets[$iCutSet]}.root ${accFileName} ${OutDirEfficiency}/Eff_times_Acc_${Particle}_Delta${CutSets[$iCutSet]}.root --batch
+      python3 CombineAccTimesEff.py ${OutDirEfficiency}/Efficiency_${Particle}_Lambda1520${CutSets[$iCutSet]}.root ${accFileName} ${OutDirEfficiency}/Eff_times_Acc_${Particle}_Lambda1520${CutSets[$iCutSet]}.root --batch
+    elif [ ${Particle} != "LctopKpi" ]; then
+      echo $(tput setaf 4) Compute efficiency times acceptance $(tput sgr0)
+      python3 CombineAccTimesEff.py ${OutDirEfficiency}/Efficiency_${Particle}${CutSets[$iCutSet]}.root ${accFileName} ${OutDirEfficiency}/Eff_times_Acc_${Particle}${CutSets[$iCutSet]}.root --batch
+    fi
+  done
+fi
+
+if $DoAccEffRw; then
+  for (( iCutSet=0; iCutSet<${arraylength}; iCutSet++ ));
+  do
+    if [ ${Particle} == "LctopKpi" ]; then
+      echo $(tput setaf 4) Compute re-weighted efficiency times acceptance $(tput sgr0)
+      python3 ComputeEffAccWeightedAvg.py ${OutDirEfficiency}/Eff_times_Acc_${Particle}${CutSets[$iCutSet]}.root ${OutDirEfficiency}/Eff_times_Acc_${Particle}_NonRes${CutSets[$iCutSet]}.root ${OutDirEfficiency}/Eff_times_Acc_${Particle}_KStar${CutSets[$iCutSet]}.root ${OutDirEfficiency}/Eff_times_Acc_${Particle}_Delta${CutSets[$iCutSet]}.root ${OutDirEfficiency}/Eff_times_Acc_${Particle}_Lambda1520${CutSets[$iCutSet]}.root ${OutDirEfficiency}/Eff_times_Acc_${Particle}${CutSets[$iCutSet]}_rw.root
```

Comment:
For loop also here

Reply:
No, here the loop cannot be used. It is just one single command line.